helm/external-dns-app/templates/crd/np.yaml

```diff
@@ -1,31 +1,4 @@
 {{- if .Values.crd.install }}
-  {{- if .Capabilities.APIVersions.Has "cilium.io/v2" }}
-apiVersion: cilium.io/v2
-kind: CiliumNetworkPolicy
-metadata:
-  name: {{ .Release.Name }}-crd-install
-  namespace: {{ .Release.Namespace | quote }}
-  annotations:
-    # create hook dependencies in the right order
-    "helm.sh/hook-weight": "-7"
-    {{- include "annotations.crd" . | nindent 4 }}
-  labels:
-    {{- include "external-dns.labels" . | nindent 4 }}
-    app.kubernetes.io/component: crd-install
-spec:
-  endpointSelector:
-    matchLabels:
-      {{- include "external-dns.selectorLabels" . | nindent 6 }}
-      app.kubernetes.io/component: crd-install
-  egress:
-    - toEntities:
-        - kube-apiserver
-      toPorts:
-        - ports:
-            - port: "443"
-        - ports:
-            - port: "6443"
-  {{- else}}
```

Comment:
this else was here to install one policy or the other one. I don't think makes sense to keep them both, as they are kind of doing the same thing.

Reply:
it still makes sense IMHO.
It is true that both policies have the same behaviour/meaning, but the `cilium` one is being added only because there is an upstream bug that prevents our use case from being addressable using k8s network policies only.
We would have preferred to avoid this vendor lock-in but we just can't.

An alternative solution would be to get rid of the cilium network policy and have an allow-all egress policy for the crd job. since this is a temporary job, the tradeoff in terms of security should not be too much.

WDYT? would that be better?

Reply:
> upstream bug that prevents our use case from being addressable using k8s network policies only

ah I see, I was not aware of that. I'm fine with keeping both until we resolve the vendor lock-in in the future